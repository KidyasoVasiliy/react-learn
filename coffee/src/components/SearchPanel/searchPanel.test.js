import React from 'react';
import { SearchPanel } from './index';
import { shallow } from 'enzyme';

describe('Testing <SearchPanel />', () => {
  it('SearchPanel have renderer correctly', () => {
    const searchPanel = shallow(<SearchPanel />);
    expect(searchPanel).toMatchSnapshot();
  })
});
