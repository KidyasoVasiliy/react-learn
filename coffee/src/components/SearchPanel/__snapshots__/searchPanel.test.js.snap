// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchPanel /> SearchPanel have renderer correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="searchpanel__label"
        >
          Lookiing for
        </div>,
        <input
          className="searchpanel__input"
          placeholder="start typing here..."
          type="text"
        />,
      ],
      "className": "searchpanel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lookiing for",
          "className": "searchpanel__label",
        },
        "ref": null,
        "rendered": "Lookiing for",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "searchpanel__input",
          "onChange": undefined,
          "placeholder": "start typing here...",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="searchpanel__label"
          >
            Lookiing for
          </div>,
          <input
            className="searchpanel__input"
            placeholder="start typing here..."
            type="text"
          />,
        ],
        "className": "searchpanel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lookiing for",
            "className": "searchpanel__label",
          },
          "ref": null,
          "rendered": "Lookiing for",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "searchpanel__input",
            "onChange": undefined,
            "placeholder": "start typing here...",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
